dotnet/example_code/ec2/Ec2SpotCRUD.cs

```diff
@@ -0,0 +1,220 @@
+//snippet-sourcedescription:[Ec2SpotCRUD demonstrates how to create, cancel, and terminate a spot instance request using Amazon EC2.]
+//snippet-keyword:[dotnet]
+//snippet-keyword:[.NET]
+//snippet-sourcesyntax:[.net]
+//snippet-keyword:[Code Sample]
+//snippet-keyword:[Amazon Elastic Compute Cloud]
+//snippet-keyword:[Amazon EC2]
+//snippet-service:[ec2]
+//snippet-sourcetype:[full-example]
+//snippet-sourcedate:[Sep 20, 2019]
+//snippet-sourceauthor:[Doug-AWS]
+/*******************************************************************************
+* Copyright 2009-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License"). You may
+* not use this file except in compliance with the License. A copy of the
+* License is located at
+*
+* http://aws.amazon.com/apache2.0/
+*
+* or in the "license" file accompanying this file. This file is
+* distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied. See the License for the specific
+* language governing permissions and limitations under the License.
+*******************************************************************************/
+/* Build 
```

Comment:
Separate README or more information about the purpose of this code--not just how to build it?

Reply:
I've added a bunch of new comments, hopefully that make it more clear what each method docs.